### PR TITLE
trigger service shutdown during upgrade

### PR DESCRIPTION
**Pull request by MaximilianMeister**

### CODE REVIEW

**Comment by rhafer** [comment]
The  calls to ```installer/upgrade/repos``` and  ```installer/upgrade/backup``` seem to be missing as well, right?

**Comment by MaximilianMeister** [comment]
> The calls to installer/upgrade/repos and installer/upgrade/backup seem to be missing as well, right?

@rhafer I would like to add them in another PR, as backup is optional, and the repo check is not expected to fail for now. would that be fine with you?
but I agree we need to test it.

This PR just fixes a chunk of the upgrade CI.

**Comment by MaximilianMeister** [comment]
> as backup is optional

having a closer look, is it really optional?

```ruby
    def backup
      ...
            @service_object.finalize_openstack_shutdown
            Openstack::Upgrade.unset_db_synced
```

I thought it is just creating a backup if wanted, but I am not so sure anymore

**Comment by rhafer** [comment]
> having a closer look, is it really optional?

No it's not. Actually it does some pretty important steps for the rest of the upgrade process.

**Comment by MaximilianMeister** [comment]
> No it's not. Actually it does some pretty important steps for the rest of the upgrade process

@rhafer updated

**Comment by rhafer** [comment]
+1 (But I modified the gating job to try a 5 -> 6 upgrade. Let's see how that goes now)

**Comment by MaximilianMeister** [comment]
>  I modified the gating job to try a 5 -> 6 upgrade. Let's see how that goes now

thanks, you were quicker :) I have done that as well for the old run https://ci.suse.de/job/openstack-mkcloud/20528/console (but the second commit was still missing)

**Comment by MaximilianMeister** [comment]
@rhafer you were missing `testsetup` and `upgrade_cloudsource` it seems. will trigger again

**Comment by rhafer** [comment]
Argh, thanks!

File: scripts/mkcloud [hunk 1]
```diff
@@ -950,6 +950,8 @@ function cloudupgrade()
 {
     if iscloudver 5plus ; then
         crowbarupgrade_5plus
+        onadmin crowbar_shutdown_services
+        onadmin crowbar_openstack_backup
```
Comment (jdsn) [inline scripts/mkcloud]: these lines should go to the `crowbarupgrade_5plus`  function, as they are part of the crowbarupgrade

Comment (rhafer) [inline scripts/mkcloud]: Hm, that depends on what you mean with "crowbarupgrade". If that refers to "Upgrading the Admin Node" then these steps are not part of the crowbarupgrade. They perform actions on the client nodes (e.g. stopping openstack and creating a backup of the postgresql database). In that case I'd argue they are more related to the nodeupgrade step and should stay together with that.

But maybe I am misunderstanding what you mean by "crowbarupgrade".

Comment (MaximilianMeister) [inline scripts/mkcloud]: I agree that needs some clarification. IMHO if the function calls belong to `crowbarupgrade_5plus()` then why isn't the call of `onadmin proposal` there as well? I see it as part of the upgrade as well.

**Comment by MaximilianMeister** [comment]
rebased

@rhafer +1 still valid ?
@jdsn added the change

**Comment by MaximilianMeister** [comment]
gating would likely be fixed by https://github.com/crowbar/crowbar-openstack/pull/354

**Comment by rhafer** [comment]
Yeah, still +1.

**Comment by rsalevsky** [comment]
+1

**Comment by jdsn** [comment]
:+1:

**Comment by vmoravec** [comment]
Upgrade scenario on qa2 succeeded https://ci.suse.de/view/Cloud/view/QA/job/cloud-mkphyscloud-qa-scenario-0a/78/console, but nodes are left in upgrade status even after barclamps have been reapplied: http://crowbar.qa2.cloud.suse.de/

**Comment by MaximilianMeister** [comment]
@vmoravec somehow already the restore failed

```
18:50:03 ++ curl -X POST -F file=@/tmp/backup-crowbar.tar.gz -s -o upgrade-start.txt -w '%{http_code}' http://localhost:3000/installer/upgrade/start.json
18:50:04 + local http_code=406
18:50:04 + [[ 406 =~ [23].. ]]
18:50:04 + cat upgrade-start.txt
18:50:04 {"status":"406","error":"Not Acceptable"}+ complain 36 'Could not start crowbar restore with upgrade workflow'
```
do you think it was due to the shell script in the job?

**Comment by MaximilianMeister** [comment]
when the backport https://github.com/crowbar/crowbar-openstack/pull/355 is merged we can trigger this gating again

**Comment by MaximilianMeister** [comment]
triggered the gating again as https://github.com/crowbar/crowbar-openstack/pull/355 has been merged

**Comment by MaximilianMeister** [comment]
```
17:10:42 + scp d52-54-77-77-77-01.vg2.cloud.suse.de:/var/lib/openstack-tempest-test/tempest.subunit.log .
17:10:42 scp: /var/lib/openstack-tempest-test/tempest.subunit.log: No such file or directory
```

this caused gating to fail. likely not related to this patch

**Comment by jdsn** [comment]
gating failure is unrelated

**Pull request merged**